src/main/java/repo/impl/PostgresMigratedFormSessionRepo.java

```
@@ -0,0 +1,418 @@
+    private String host;
+
+    @Autowired
+    private RestoreFactory restoreFactory;
+
+    public static final String POSTGRES_MIGRATED_SESSION_TABLE_NAME = "formplayer_session";
+    public static final String POSTGRES_MIGRATED_ENTRYSESSION_TABLE_NAME = "formplayer_entrysession";
+    public static final String POSTGRES_AUTH_USER_TABLE_NAME = "auth_user";
+
+    @Override
+    public List<SerializableFormSession> findUserSessions(String username) {
+
+        List<Integer> userIds = this.jdbcTemplate.query(
+                replaceTableName("SELECT * FROM %s WHERE username = ?", POSTGRES_AUTH_USER_TABLE_NAME),
+                new Object[] {username},
+                new UserIdMapper());
+
+        // should be only one
+        int userId = userIds.get(0);
+
+        List<SerializableFormSession> sessions = new ArrayList<>();
+
+        List<EntrySession> entrySessions = this.jdbcTemplate.query(
+                replaceTableName("SELECT * FROM %s WHERE user_id = ?", POSTGRES_MIGRATED_ENTRYSESSION_TABLE_NAME),
+                new Object[] {userId},
+                new EntrySessionMapper());
+
+        for (EntrySession entrySession: entrySessions) {
+            String sessionId = entrySession.getSessionId();
+            List<InstanceSession> instanceSessions = this.jdbcTemplate.query(
+                    replaceTableName("SELECT * FROM %s WHERE sess_id = ?", POSTGRES_MIGRATED_SESSION_TABLE_NAME),
+                    new Object[] {sessionId},
+                    new InstanceSessionMapper());
+            if(instanceSessions.size() < 1) {
+                continue;
+            }
+            sessions.add(buildSerializedSession(entrySession, instanceSessions.get(0)));
+        }
+        return sessions;
+    }
+
+    @Override
+    @Lock(LockModeType.OPTIMISTIC)
+    public <S extends SerializableFormSession> Iterable<S> save(Iterable<S> entities) {
+        for(SerializableFormSession session: entities){
+            save(session);
```

if `save` is doing nothing then why bother iterating? Also would it not be better to raise an exception here since presumably this should never get called.